Dockerfile

```diff
@@ -1,19 +1,91 @@
-FROM node:14-alpine
+# ==
+# builder
+# ==
+
+# application builder
+FROM node:16 AS builder
 
-# directory for the app in the container
 WORKDIR /usr/app
 
-# copies all the app's files from host into the container folder which
-# might include the node_modules dir if npm install executed in the host
+# copy package.json and package-lock.json files
+COPY package*.json .
+
+# install dependencies including local development tools
+RUN npm install
+
+# copy the rest of the content
```

Comment:
This copy command does not copy the node_modules folder because it's in the dockerignore file, right?

Reply:
Correct, and [as per the official docs](https://docs.docker.com/engine/reference/builder/#dockerignore-file), if the `.dockerignore` file is not present, docker ignores the files present in the `.gitignore`